### PR TITLE
Backport of CI updates to 6.0

### PR DESCRIPTION
**Pull request by ckreibich**
This is a bunch of cherry-picks to bring `release/6.0` in line with what we run in CI on `master`. It skips the following:
- The recent `-Werror` introduction
- The broader benchmarker configuration (it already runs for Ubuntu 22, though)
- Spicy special-casing via the `ubuntu22_spicy_task` and `ubuntu22_spicy_head_task` targets — these seem more appropriate for `master`

### CODE REVIEW

**Comment by timwoj** [comment]
I updated the docs submodule so it should pass now if you want to rebase.

File: .cirrus.yml [hunk 1]
```diff
@@ -96,6 +96,7 @@ ci_template: &CI_TEMPLATE
 
   init_external_repos_script: ./ci/init-external-repos.sh
 
+  pre_build_script: ./ci/pre-build.sh
```
Comment (timwoj) [inline .cirrus.yml:92]: Is this used by anything in 6.0? On master it's used in the `ubuntu22_spicy_head_task` task, but we don't have that task on 6.0 (or it wasn't added by this PR).

**Review by timwoj — CHANGES_REQUESTED** [review]
One comment on something I don't think we need. The rest looks fine to me.

File: .cirrus.yml (hunk 1 above)
Comment (ckreibich) [reply .cirrus.yml:92]: Yeah you're right, let me drop it. Thanks!

**Review by timwoj — APPROVED** [review]

**Pull request merged**